GH-130: Add missing 2.4.6 BIC highlights

jeff-matthews opened the pull request:
## Purpose of this pull request

This pull request (PR) fixes #130. It adds a functional change (disable auto redirects by default) that technically qualifies as a backward-incompatible change for 2.4.6. It also adds info about Zend_Json removal.

Jira tickets:

- ACPT-489 (disable auto redirect)
- AC-5886 (Zend_Json replacement)

## Affected pages

- https://developer.adobe.com/commerce/php/development/backward-incompatible-changes/highlights/

## Links to Magento Open Source code

- https://github.com/magento/magento2/commit/78976ed4a52045872089e235b33a6ab13913c299
- https://github.com/magento/magento2/commit/41ac34b84df4ca1e69ab93d2930b98fda5ce5087

keharper commented on src/pages/development/backward-incompatible-changes/highlights.md:
```diff
@@ -19,6 +19,12 @@ The following major backward-incompatible changes were introduced in the 2.4.6 A
+
+To improve performance, the default value for `generate_category_product_rewrites` in the [`app/code/Magento/CatalogURLRewrite/etc/config.xml`](https://github.com/magento/magento2/blob/2.4-develop/app/code/Magento/CatalogUrlRewrite/etc/config.xml#L12) file was changed from `1` to `0`. This change disables [automatic redirects](https://experienceleague.adobe.com/docs/commerce-admin/marketing/seo/u ...
+
+The new default does not change existing records in the `catalog_url_rewrite_product_category` and `url_rewrite` database tables when upgrading to 2.4.6, but no new rewrites are added. However, manually changing this configuration setting after upgrading to 2.4.6 permanently deletes all rewrites with no ability to restore them. This may cause unresolved "category/product" type URL conflicts that  ...
```
```suggestion
The new default does not change existing records in the `catalog_url_rewrite_product_category` and `url_rewrite` database tables when upgrading to 2.4.6, but no new rewrites are added. However, using the `bin/magento config:set catalog/seo/generate_category_product_rewrites command to change this configuration setting after upgrading to 2.4.6 permanently deletes all rewrites with no ability to restore them. This may cause unresolved "category/product" type URL conflicts that you must resolve by updating URL keys manually.
```

keharper reviewed (commented):
I recommend stating that the value of Stores > Configuration > Catalog > Catalog > Search Engine Optimization > Generate "category/product" URL Rewrites did not change. 

You should probably add this item to the [Catalog configuration paths reference](https://experienceleague.adobe.com/docs/commerce-operations/configuration-guide/paths/config-reference-catalog.html?lang=en)

catalog/seo/generate_category_product_rewrites

All of this pending approval from Jacob.

jfrontain reviewed (approved):
approving but please incorporate keharper's incisive suggestion

jeff-matthews commented:
> I recommend stating that the value of Stores > Configuration > Catalog > Catalog > Search Engine Optimization > Generate "category/product" URL Rewrites did not change.
> 

@JacobBrownAustin, can you please confirm?

JacobBrownAustin commented on src/pages/development/backward-incompatible-changes/highlights.md:
```diff
@@ -11,14 +11,22 @@ This page highlights backward-incompatible changes between Adobe Commerce and Ma
 
+### New default value for automatic redirects
+
+To improve performance, the default value for `generate_category_product_rewrites` in the [`app/code/Magento/CatalogURLRewrite/etc/config.xml`](https://github.com/magento/magento2/blob/2.4-develop/app/code/Magento/CatalogUrlRewrite/etc/config.xml#L12) file was changed from `1` to `0`. This change disables [automatic redirects](https://experienceleague.adobe.com/docs/commerce-admin/marketing/seo/u ...
```
> This change disables automatic redirects

It's for rewrites, not redirects.  But only for rewrites with respect to the "catalog/product".

Though the weird thing is that even without the redirect,  the page still loads on my dev environment.
For example, on my local development environment, I've created subcategory "subcat2" and product "simpleproduct2" inside this category.
I can still access this new product via http://nginx.docker/index.php/subcat2/simpleproduct2.html or http://nginx.docker/subcat2/simpleproduct2.html even though I've disabled the generate_category_product_rewrites and the rewrite for this doesn't exist in `catalog_url_rewrite_product_category` and `url_rewrite`.

jeff-matthews replied:
Ah, yes. Thanks for correcting me on that. I'll change the link text to "rewrites" and link to the [proper section](https://experienceleague.adobe.com/docs/commerce-admin/marketing/seo/url-rewrites/url-redirect-product-automatic.html#skip-rewrite) in docs.

JacobBrownAustin commented on src/pages/development/backward-incompatible-changes/highlights.md:
```diff
@@ -11,14 +11,22 @@ This page highlights backward-incompatible changes between Adobe Commerce and Ma
+To improve performance, the default value for `generate_category_product_rewrites` in the [`app/code/Magento/CatalogURLRewrite/etc/config.xml`](https://github.com/magento/magento2/blob/2.4-develop/app/code/Magento/CatalogUrlRewrite/etc/config.xml#L12) file was changed from `1` to `0`. This change disables [automatic redirects](https://experienceleague.adobe.com/docs/commerce-admin/marketing/seo/u ...
+
+The new default does not change existing records in the `catalog_url_rewrite_product_category` and `url_rewrite` database tables when upgrading to 2.4.6, but no new rewrites are added. However, using the `bin/magento config:set catalog/seo/generate_category_product_rewrites` command to change this configuration setting after upgrading to 2.4.6 permanently deletes all rewrites with no ability to r ...
+
```
We should add suggestion that if the generate_category_product_rewrites functionality is wanted for them,  they should manually set this value, and it will work with same functionality as before.

Here's example of what I think we should add.

> If `Generate "category/product" URL Rewrites` functionality is still wanted, it is recommended to manual set this configuration option to enabled, such as `bin/magento config:set catalog/seo/generate_category_product_rewrites 1`, then behaviour will stay the same after upgrade.

jeff-matthews commented:
Thanks for the reviews @keharper, @JacobBrownAustin, and @jfrontain.

I think I addressed all comments. I create a separate PR for updating the configuration path reference since that lives in a different repo (git.corp.adobe.com). 

Please review.

jeff-matthews commented on src/pages/development/backward-incompatible-changes/highlights.md:
```diff
@@ -11,14 +11,26 @@ This page highlights backward-incompatible changes between Adobe Commerce and Ma
+
+<InlineAlert variant="warning" slots="text" />
+
+Manually changing this setting (for example, using the `bin/magento config:set catalog/seo/generate_category_product_rewrites` command) permanently deletes all rewrites with no ability to restore them. This may cause unresolved category/product type URL conflicts that you must resolve by manually updating URL keys.
```
I successfully ran the command on my local docker instance:

```
vagrant@ubuntu2204:~/local-dev/docker-magento$ bin/magento config:set catalog/seo/generate_category_product_rewrites 1
Value was saved.
```
